soc/bflb/bl61x/soc.c

```diff
@@ -250,6 +229,21 @@ static void disable_interrupt_autostacking(void)
 }
 
 
+void arch_cache_init(void)
+{
+	enable_thead_isa_ext();
+	set_thead_enforce_aligned(false);
+	cache_data_enable();
+	/* branch prediction can cause major slowdowns (250ms -> 2 seconds)
+	 * in some applications
+	 */
```

Comment:
This means that when you enable branch prediction the CPU can stall up to 2s ?
What that means? This should be in the commit body with clear explanation and evidence, not here in as floating comment.

Reply:
It means that on repetitive branch prediction failures (case was micropython rv32 emitter), it can cause huge slowdown.